src/adapter/src/coord/ddl.rs

```diff
@@ -93,12 +94,21 @@ impl<S: Append + 'static> Coordinator<S> {
                             _ => {}
                         }
                     }
-                    CatalogItem::Sink(catalog::Sink {
-                        connection: SinkConnectionState::Ready(_),
-                        compute_instance,
-                        ..
-                    }) => {
-                        sinks_to_drop.push((*compute_instance, *id));
+                    CatalogItem::Sink(catalog::Sink { connection, .. }) => {
+                        match connection {
+                            StorageSinkConnectionState::Ready(_) => {
+                                storage_sinks_to_drop.push(*id);
+                            }
+                            StorageSinkConnectionState::Finalizing(_) => {
+                                storage_temporary_sinks_to_drop.push(*id);
+                            }
+                            // XXX(chae): Should we just ignore a drop if we're still pending (haven't started processing
+                            // the sink ready message)?? I'm worried that if we don't ignore, we'd have a panic if we get
```

Comment:
We would get a panic because the storage controller is being asked to drop a sink that doesn't exist (according to it's knowledge)?